crates/matrix-qrcode/Cargo.toml

```diff
@@ -22,9 +22,9 @@ decode_image = ["image", "rqrr", "qrcode/image", "qrcode/svg"]
 [dependencies]
 base64 = "0.13.0"
 byteorder = "1.4.3"
-image = { version = "0.23.0", optional = true }
+image = { version = "0.24.0", optional = true }
 qrcode = { version = "0.12.0", default-features = false }
-rqrr = { version = "0.4.0", optional = true }
+rqrr = { version = "0.5.0", optional = true }
```

Comment:
@poljar what's the problem exactly with upgrading these? They seem to build fine... Is there a runtime issue with them then? If so, shouldn't we add a regression tests to ensure we don't accidentally break these?

Reply:
They do build fine? AFAIR they at least required code changes, which the CI seems to agree with.

The problem is that the [qrcode](https://github.com/kennytm/qrcode-rust/pull/54) crate still depends on image 0.24, this was briefly explained when I bumped the deps here: https://github.com/matrix-org/matrix-rust-sdk/pull/618#issue-1217190348=

But perhaps I misunderstood what cargo told me, or I forgot to run `cargo update`.

Reply:
yes, they build fine here. with and without `cargo update`.

```
cargo build -p matrix-sdk-qrcode
   Compiling num-rational v0.4.0
   Compiling getrandom v0.2.6
   Compiling flate2 v1.0.23
   Compiling png v0.17.5
   Compiling spin v0.9.3
   Compiling image v0.23.14
   Compiling jpeg-decoder v0.2.6
   Compiling ahash v0.7.6
   Compiling nanorand v0.7.0
   Compiling hashbrown v0.11.2
   Compiling flume v0.10.12
   Compiling tiff v0.7.2
   Compiling exr v1.4.2
   Compiling indexmap v1.8.1
   Compiling lru v0.7.5
   Compiling qrcode v0.12.0
   Compiling ruma-common v0.9.2
   Compiling image v0.24.2
   Compiling rqrr v0.5.1
   Compiling matrix-sdk-qrcode v0.3.0 (/home/ben/Dev/matrix/matrix-rust-sdk/crates/matrix-sdk-qrcode)
    Finished dev [unoptimized + debuginfo] target(s) in 8.96s
[ben@slimbook matrix-rust-sdk]$ cargo update
    Updating crates.io index
    Updating clap v3.1.17 -> v3.1.18
    Updating clap_derive v3.1.7 -> v3.1.18
    Updating ed25519 v1.4.1 -> v1.5.0
    Updating indoc v1.0.5 -> v1.0.6
    Updating syn v1.0.92 -> v1.0.94
    Updating tokio v1.18.1 -> v1.18.2
    Removing unindent v0.1.8
[ben@slimbook matrix-rust-sdk]$ cargo build -p matrix-sdk-qrcode
   Compiling syn v1.0.94
   Compiling synstructure v0.12.6
   Compiling serde_derive v1.0.137
   Compiling thiserror-impl v1.0.31
   Compiling zeroize_derive v1.3.2
   Compiling pin-project-internal v1.0.10
   Compiling prost-derive v0.10.1
   Compiling tracing-attributes v0.1.21
   Compiling zeroize v1.3.0
   Compiling pin-project v1.0.10
   Compiling flume v0.10.12
   Compiling tracing v0.1.34
   Compiling prost v0.10.3
   Compiling exr v1.4.2
   Compiling thiserror v1.0.31
   Compiling ruma-identifiers-validation v0.8.1
   Compiling serde v1.0.137
   Compiling image v0.24.2
   Compiling toml v0.5.9
   Compiling curve25519-dalek v3.2.1
   Compiling serde_bytes v0.11.6
   Compiling serde_json v1.0.81
   Compiling ed25519 v1.5.0
   Compiling arrayvec v0.7.2
   Compiling js_option v0.1.0
   Compiling indexmap v1.8.1
   Compiling js_int v0.2.2
   Compiling rqrr v0.5.1
   Compiling x25519-dalek v1.2.0
   Compiling ed25519-dalek v1.0.1
   Compiling proc-macro-crate v1.1.3
   Compiling vodozemac v0.2.0
   Compiling ruma-macros v0.9.2
   Compiling ruma-common v0.9.2
   Compiling matrix-sdk-qrcode v0.3.0 (/home/ben/Dev/matrix/matrix-rust-sdk/crates/matrix-sdk-qrcode)
    Finished dev [unoptimized + debuginfo] target(s) in 50.87s
    
```

But tests seem to fail. Will check.

Reply:
I see what you mean, though checking the rust-qrcode crate's repo leaves me concerned this isn't maintained anymore. But yeah, looks like this isn't going to work without a release of qrcode... reverting.

Reply:
You'll need to enable the relevant features, as you can see from your diff `rqrr` is optional.

Reply:
> I see what you mean, though checking the rust-qrcode crate's repo leaves me concerned this isn't maintained anymore. But yeah, looks like this isn't going to work without a release of qrcode... reverting.

Indeed, not sure what we should do about it.